spaceship/lib/spaceship/du/du_client.rb

```diff
@@ -40,6 +44,30 @@ def upload_trailer_preview(app_version, upload_file, content_provider_id, sso_to
 
     private
 
+    def upload_purchase_preview(app_id, upload_file, path, content_provider_id, sso_token, du_validation_rule_set = "MZPFT.SortedScreenShot")
+      referrer = "https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa/ra/ng/app/#{app_id}/addons/create/consumable"
+      r = request(:post) do |req|
+        req.url "#{self.class.hostname}#{path}"
+        req.body = upload_file.bytes
+        req.headers['Accept'] = 'application/json, text/plain, */*'
+        req.headers['Content-Type'] = upload_file.content_type
+        req.headers['X-Apple-Upload-Referrer'] = referrer
+        req.headers['Referrer'] = referrer
+        req.headers['X-Apple-Upload-AppleId'] = app_id
+        req.headers['X-Apple-Upload-itctoken'] = sso_token
+        req.headers['X-Apple-Upload-ContentProviderId'] = content_provider_id
+        req.headers['X-Original-Filename'] = upload_file.file_name
+        req.headers['X-Apple-Upload-Validation-RuleSets'] = du_validation_rule_set if du_validation_rule_set
+        req.headers['Content-Length'] = upload_file.file_size.to_s
+        req.headers['Connection'] = "keep-alive"
+      end
+      if r.status == 500 and r.body.include?("Server Error")
+        return upload_purchase_preview(app_id, upload_file, path, content_provider_id, sso_token, du_validation_rule_set)
```

Comment:
done, in fact this has been taken from `upload_image` - but to not break something, i would like to keep it there, and do a seperate PR for the `upload_image` to also stop after n trys.